src/ch00/interacting/interfaces-traits.md

```diff
@@ -6,12 +6,9 @@ In cairo, to specify the interface you need to define a trait annotated with `#[
 
 When a function needs to access the contract state, it must have a `self` parameter of type `ContractState`. This implies that the corresponding function signature in the interface trait must also take a `TContractState` type as a parameter. It's important to note that every function in the contract interface must have this `self` parameter of type `TContractState`.
 
-You can use the `#[generate_trait]` attribute to implicitly generate the trait for a specific implementation block. This attribute automatically generates a trait with the same functions as the ones in the implemented block, replacing the `self` parameter with a generic `TContractState` parameter.
+You can use the `#[generate_trait]` attribute to implicitly generate the trait for a specific implementation block. This attribute automatically generates a trait with the same functions as the ones in the implemented block, replacing the `self` parameter with a generic `TContractState` parameter. However, this will only work for internal functions, and not for functions that are part of the contract's public interface - as embedding an impl inside the abi of a contract requires this impl to implement an interface.
```

Comment:
See my comment in #125 (I think that `external(v0)` will be deprecated on the impl only)
Let's consider a contract with:
```rust
#[abi(per_item)]
#[generate_trait]
impl Contract of IContract {
    #[external(v0)]
    fn a(self: @ContractState) -> u32 {
        0
    }
}
```

Even with no explicit interface you can still declare your own dispatcher and use it, like this:
```rust
mod tests {
    #[starknet::interface]
    trait IGlobalExampleA<TContractState> {
        fn a(self: @TContractState) -> u32;
    }

    #[test]
    #[available_gas(2000000000)]
    fn test_a() {
        let (address0, _) = starknet::deploy_syscall(
            my_contracts::Contract::TEST_CLASS_HASH.try_into().unwrap(), 0, ArrayTrait::new().span(), false
        ).unwrap();

        let contract = IGlobalExampleADispatcher { contract_address: address0 };
        assert(contract.a() == 1, 'contract.(a) != 1');
    }
}
```

Reply:
However I do agree that it's a best practice to always have one external functions block using `#[abi(embed_v0)]` than implement an explicitly defined interface, but it's not mandatory. Correct me if I'm wrong!

Reply:
No I think you're right!